src/main/java/net/pms/formats/v2/AudioProperties.java

```diff
@@ -166,14 +226,48 @@ public static int getChannelsNumberFromLibMediaInfo(String mediaInfoValue) {
 		}
 	}
 
+	public static int getBitsperSampleFromLibMediaInfo(String mediaInfoValue) {
+		if (isEmpty(mediaInfoValue)) {
+			LOGGER.warn("Empty value passed in. Returning default number 16.");
+			return 16;
+		}
+
+		// examples of libmediainfo  (mediainfo --Full --Language=raw file):
+		// Bit depth : 16
+		// Bit depth : 24
+		// Bit depth : / 24 / 24
+
+		int result = -1;
+		Matcher intMatcher = intPattern.matcher(mediaInfoValue);
+		while (intMatcher.find()) {
```

Comment:
@Sami32 The thing is that you're doing it in a loop. Regex will look from left to right until it finds an integer number. When that happens, to loop will run the first time, setting the leftmost number in ```result```. But, since you use ```while``` it will keep going, so regex will look again - from the position is stopped last. If there are more integer numbers in the string, it will get another "hit" and run through your code again. This will overwrite the value already stored in ```result```, and this will repeat itself until there are no more integer numbers in the string. When regex can't find any more integer numbers, the loop will exit and the code continue. The last value set in ```result``` will be the rightmost integer number in the string.

This is a very inefficient way to parse a single number, because you basicly do some "work" (parsing the string etc) which you then discards. If you really want the rightmost number the regex could be modified to actually find this in the first place, and you wouldn't need to use a loop. But, since we're talking only a few numbers here, I doubt that the "cost" of this will be noticeable. My "instinct" is not to do unnecessary things though.